tests/queries/0_stateless/02907_backup_mv_with_no_inner_table.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+${CLICKHOUSE_CLIENT} -nm --query "
+drop table if exists src;
+create table src (a Int32) engine = MergeTree() order by tuple();
+
+drop table if exists mv;
+create materialized view mv (a Int32) engine = MergeTree() order by tuple() as select * from src;
+"
+
+uuid=$(${CLICKHOUSE_CLIENT} --query "select uuid from system.tables where table='mv' and database == currentDatabase()")
+inner_table=".inner_id.${uuid}"
+${CLICKHOUSE_CLIENT} -nm --query "drop table \`$inner_table\` sync"
+
+${CLICKHOUSE_CLIENT} -nm --query "
+backup table ${CLICKHOUSE_DATABASE}.\`mv\` on cluster test_shard_localhost to Disk('backups', '${CLICKHOUSE_TEST_UNIQUE_NAME}');
+" | grep -o "BACKUP_CREATED"
+
+${CLICKHOUSE_CLIENT} -nm --query "
+drop table mv;
+restore table ${CLICKHOUSE_DATABASE}.\`mv\` on cluster test_shard_localhost from Disk('backups', '${CLICKHOUSE_TEST_UNIQUE_NAME}');
+" | grep -o "RESTORED"
+
+${CLICKHOUSE_CLIENT} --query "select count() from mv;"
```

Comment:
As far as I know `SELECT` from a materialized view with its internal table dropped should throw an exception, shouldn't it?

Reply:
yes, but after restore it started to work again so I assume empty inner table was created as a result of a restore.

Reply:
ok